spec/models/financial_transaction_spec.rb

```diff
@@ -0,0 +1,31 @@
+require_relative '../spec_helper'
+
+describe FinancialTransaction do
+  let(:ordergroup) { create(:ordergroup) }
+  let(:ft) { create(:financial_transaction, ordergroup: ordergroup, amount: 20) }
+
+  before do
+    ft
+    ordergroup
+  end
+
+  it 'updates the amount of the ordergroup balance' do
+    expect(ordergroup.account_balance).to eq(20)
+    create(:financial_transaction, ordergroup: ordergroup, amount: 10)
+    expect(ordergroup.account_balance).to eq(30)
+  end
+
+  it 'can be reverted' do
+    ft.revert!(ft.user)
+    expect(ft).to be_hidden
+    expect(ordergroup.financial_transactions.count).to eq 2
+    expect(ordergroup.financial_transactions.last.amount).to eq(-20)
+    expect(ordergroup.financial_transactions.last.reverts).to eq ft
+    expect(ordergroup.account_balance).to eq 0
+  end
+
+  it 'fails when trying to revert a pending transaction (amount=nil)' do
+    ft.update_attribute(:amount, nil)
```

Comment:
I'd suggest to create a financial transaction trait in the factory that makes it more expressive, and will allow it to function when we'd change behaviour (e.g. have an explicit _pending_ attribute or so).

```ruby
context "with a pending transaction" do
  let!(:ft) { create(:financial_transaction, :pending, ordergroup: ordergroup, amount: 20) }
  it "fails on revert" do
      expect { ft.revert!(ft.user) }.to raise_error(RuntimeError, 'Pending Transaction cannot be reverted')
  end
end
```